### DISCUSSION TITLE
Sub Button Icon - change with entity state?

### DISCUSSION THREAD

**Discussion by ExR90** [Clooos/Bubble-Card#557] [Questions about config, custom styles and templates] [answered]
I just started using the Bubble Cards. With the sub-buttons, the icon picks up from the sub-button defined entity but does not change with the status. Example being a lock, which normally updates the icon to show unlocked. It does not do this with the sub-buttons. It just shows the icons background color getting bright/dark based on entity status.

Would love the sub-button icon changes as it would if it was a "primary" button vs a sub-button.

**Comment by Clooos** [comment, marked as answer]
Hi, I know that they are still some default icons that are still not automatically changing, this is because I have to add manually all the cases.

In the meantime you can still create a template to change that icon based on the state, you can find a lot of examples in the documentation :)

**Reply by mooseBringer** [reply]
Came here to ask a similar question. I'm trying to light up one of four buttons depending on an `input_text` but can't get it to work. If I replace the if statement it changes color and the if statement works in a normal button with card_mod but not here. I think the syntax may have to be different but can't figure it out.

```
type: custom:bubble-card
card_type: button
entity: input_boolean.kylaauto
button_type: switch
show_state: true
name: Auto
sub_button:
  - entity: script.ac_kyla_full
    show_name: true
    show_state: false
    name: Full
    tap_action:
      action: toggle
    show_icon: true
    icon: mdi:fan-plus
  - entity: script.ac_kyla_halv
    name: Halv
    show_name: true
    tap_action:
      action: toggle
    show_icon: true
    icon: mdi:fan
  - entity: script.ac_kyla_natt
    name: Natt
    show_name: true
    tap_action:
      action: toggle
    show_icon: true
    icon: mdi:fan-minus
  - entity: script.ac_kyla_mjukav
    name: Av*
    show_name: true
    tap_action:
      action: toggle
    show_icon: true
    icon: mdi:fan-off
    hold_action:
      action: call-service
      target:
        entity_id: script.ac_kyla_av
card_layout: normal
styles: |
  .bubble-button-background {
    background: paleblue !important;
    opacity: 0.4 !important;
  }
  .bubble-sub-button-0 {
    background-color: {% if is_state('input_text.sovrumsac_valtlage', 'Full') %} limegreen {% else %} slategray {% endif %} !important;
  }
  .bubble-sub-button-1 {
    background-color: {% if is_state('input_text.sovrumsac_valtlage', 'Halv') %} limegreen {% else %} slategray {% endif %} !important;
  }
  .bubble-sub-button-2 {
    background-color: {% if is_state('input_text.sovrumsac_valtlage', 'Natt') %} limegreen {% else %} slategray {% endif %} !important;
  }
  .bubble-sub-button-3 {
    background-color: {% if is_state('input_text.sovrumsac_valtlage', 'Av') %} limegreen {% else %} slategray {% endif %} !important;
  } 

```

**Reply by Clooos** [reply]
Bubble Card doesn't support Jinja templates, only JS templates, but I've added many examples in the documentation.

**Reply by mooseBringer** [reply]
Ok that explains it, will check it out.

Also thank you! Amazing work! 💪🏻

**Reply by mooseBringer** [reply]
Ah yes! Thank you! Now I found the right one

```
  .bubble-sub-button-4 {
    background-color: ${hass.states['input_text.sovrumsac_valtlage'].state === 'Av' ? 'green' : 'grey'} !important; 
  } 
```

**Reply by ExR90** [reply]
> Hi, I know that they are still some default icons that are still not automatically changing, this is because I have to add manually all the cases.
> 
> In the meantime you can still create a template to change that icon based on the state, you can find a lot of examples in the documentation :)

Oh I am aware you can do it via templates, my question was more if the inconsistent functionality without using templates was intentional etc.

**Reply by Clooos** [reply]
I've forgot to add the lock icons indeed, but I've added a lot of new icons in the v2. Don't hesitate to open an issue if you find more icons that are not changing, and I will add them in an upcoming release.
